test/AbstractPluginManagerTest.php

```
@@ -349,4 +350,17 @@ public function testPassingServiceInstanceViaConfigureShouldRaiseExceptionForInv
             stdClass::class => new stdClass(),
         ]]);
     }
+
+    public function testAbstractFactoryGetsCreationContext()
+    {
+        $serviceManager = new ServiceManager();
+        $pluginManager = new TestAsset\SimplePluginManager($serviceManager);
+        $abstractFactory = $this->prophesize(AbstractFactoryInterface::class);
+        $abstractFactory->canCreate($serviceManager, 'foo')
+            ->willReturn(true);
+        $abstractFactory->__invoke($serviceManager, 'foo', null)
+            ->willReturn(new InvokableObject());
```

`shouldBeCalled()` is not mandatory here, but it is useful for debugging. If the method does not get called, then I'd prefer the test to fail early telling that it was not called instead instad of failing on `assertInstanceOf()` where I can't see why it failed.